sdk/src/main/java/io/dapr/client/domain/StateOptions.java

```diff
@@ -29,6 +33,31 @@ public RetryPolicy getRetryPolicy() {
     return retryPolicy;
   }
 
+  public Map<String, String> getStateOptionsAsMap() {
+    Map<String, String> mapOptions = null;
+    if (this != null) {
+      mapOptions = new HashMap<>();
+      if (this.getConsistency() != null) {
+        mapOptions.put("consistency", this.getConsistency().getValue());
+      }
+      if (this.getConcurrency() != null) {
+        mapOptions.put("concurrency", this.getConcurrency().getValue());
+      }
+      if (this.getRetryPolicy() != null) {
+        if (this.getRetryPolicy().getInterval() != null) {
+          mapOptions.put("retryInterval", DurationUtils.ConvertDurationToDaprFormat(this.getRetryPolicy().getInterval()));
+        }
+        if (this.getRetryPolicy().getThreshold() != null) {
+          mapOptions.put("retryThreshold", this.getRetryPolicy().getThreshold().toString());
+        }
+        if (this.getRetryPolicy().getPattern() != null) {
+          mapOptions.put("retryPattern", this.getRetryPolicy().getPattern().getValue());
+        }
+      }
+    }
+    return mapOptions;
```

Comment:
return Collections.unmodifiableMap(mapOptions);

Reply:
done, can you look at it again?